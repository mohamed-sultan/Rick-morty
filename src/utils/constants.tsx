import RocketLaunchIcon from "@mui/icons-material/RocketLaunch";

export const SideMenuLinks = [
  {
    id: 1,
    title: "Now Playing",
    path: "/",
    icon: <RocketLaunchIcon color="primary" sx={{ fontSize: 40 }} />,
  },
  {
    id: 2,
    title: "Popular",
    path: "/categories/popular",
    icon: <RocketLaunchIcon color="primary" sx={{ fontSize: 40 }} />,
  },
  {
    id: 3,
    title: "Top Rated",
    path: "/categories/top-rated",
    icon: <RocketLaunchIcon color="primary" sx={{ fontSize: 40 }} />,
  },
  {
    id: 4,
    title: "Upcoming",
    path: "/categories/upcoming",
    icon: <RocketLaunchIcon color="primary" sx={{ fontSize: 40 }} />,
  },
];
