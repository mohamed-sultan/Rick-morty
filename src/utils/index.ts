export const saveAppearance = (isDark: Boolean) => {
  localStorage.setItem("dark", isDark ? "true" : "false");
};
export const saveInitialAppearance = () => {
  const isDark = localStorage.getItem("dark") === "true";
  return isDark;
};
