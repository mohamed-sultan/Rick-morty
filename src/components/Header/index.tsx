import { styled } from "@mui/material/styles";
import { useDispatch, useSelector } from "react-redux";
import { selectDarkState, setDarkState } from "../../redux/dark";
import BedtimeIcon from "@mui/icons-material/Bedtime";
import IconButton from "@mui/material/IconButton";
import { saveAppearance } from "../../utils";

const TitleText = styled("span")(({ theme }) => ({
  color: theme.palette.primary?.purple,
  fontSize: "1.6rem",
}));
const Header = styled("div")(({ theme }) => ({
  borderBottom: `1px solid ${theme.palette.primary?.silver}`,
  padding: "1.7rem ",
  color: theme.palette.primary?.black,
  display: "flex",
  justifyContent: "space-between",
  alignItems: "center",
  position: "fixed",
  top: 0,
  left: 0,
  right: 0,
  zIndex: 1000,
  backgroundColor: theme.palette.primary?.white,
}));

const Head = () => {
  const dispatch = useDispatch();
  const isDark = useSelector(selectDarkState);
  return (
    <Header>
      <TitleText>Rick and Morty </TitleText>
      <IconButton
        aria-label="dark-light-icon"
        onClick={() => {
          dispatch(setDarkState(!isDark));
          saveAppearance(!isDark);
        }}
      >
        <BedtimeIcon sx={{ fontSize: 40, color: "primary.red" }} />
      </IconButton>
    </Header>
  );
};

export default Head;
