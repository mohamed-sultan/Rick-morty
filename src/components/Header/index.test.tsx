import * as React from "react";
import TestRenderer from "react-test-renderer";
import Header from "./index";
import { saveAppearance } from "../../utils";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

import { Provider } from "react-redux";
import ConfigureStore from "redux-mock-store";
describe("Header snapshot testing", () => {
  jest.spyOn(Object.getPrototypeOf(localStorage), "getItem");
  jest.spyOn(Object.getPrototypeOf(localStorage), "setItem");
  const mockStore = ConfigureStore([]);
  const store = mockStore({ ...{ dark: { dark: true } } });
  const Component = () => (
    <Provider store={store}>
      <Header />
    </Provider>
  );
  it("renders correctly", () => {
    const tree = TestRenderer.create(<Component />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("click the icon button should call the localstorage", () => {
    const { getByLabelText } = render(<Component />);
    const buttonIcon = getByLabelText("dark-light-icon");
    expect(buttonIcon).toBeInTheDocument();
    fireEvent.click(buttonIcon);
    expect(localStorage.setItem).toHaveBeenCalled();
  });
});
