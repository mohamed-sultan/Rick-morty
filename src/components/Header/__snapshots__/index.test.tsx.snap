// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot testing renders correctly 1`] = `
<div
  className="css-1e0kmke"
>
  <span
    className="css-1nedsoq"
  >
    Rick and Morty 
  </span>
  <button
    aria-label="dark-light-icon"
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-9sdrmz-MuiSvgIcon-root"
      data-testid="BedtimeIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12.34 2.02C6.59 1.82 2 6.42 2 12c0 5.52 4.48 10 10 10 3.71 0 6.93-2.02 8.66-5.02-7.51-.25-12.09-8.43-8.32-14.96z"
      />
    </svg>
  </button>
</div>
`;
